lib/eco/export/jruby.py

```diff
@@ -381,3 +389,63 @@ def _profile(self):
                              num_calls))
                 self._annotate_text(method.source.line_start, method.name,
                                     dmsg, JRubyMorphismMsg)
+                # Find and annotate method calls.
+                for version in method.versions:
+                    for callsite_version in version.called_from:
+                        if method.is_mega:
+                            cmsg = ('Call to %s defined on line %g (megamorphic).' %
+                                    (method.name, method.source.line_start))
+                        else:
+                            cmsg = ('Call to %s defined on line %g.\n' %
+                                    (method.name, method.source.line_start))
+                        cmsg += '\n(line numbers may be inaccurate in polyglot programs)'
+                        self._annotate_text(callsite_version.callsite.line,
+                                            method.name, cmsg, JRubyMorphismMsg)
+                        if callsite_version.callsite.line > 0:
+                            self._annotate_text(callsite_version.callsite.line,
+                                                method.name,
+                                                { method.name : method.is_mega },
+                                                JRubyMorphismLine)
+                        num_calls += 1
+
+    def _annotate_text(self, lineno, text, annotation, klass):
+        """Annotate a node on a given line with a given symbol name.
+        """
+        if lineno < 0:
```

Comment:
This happens when method is called which has not been defined by the programmer. For example,  in the snippet:

```ruby
10.times do
    puts "hello"
end
```

`times` is a method on the object `10`, but the user has not defined `times`, so according to the callgraph log, the source section from which `times` is defined starts on line -1. Below is an example of (part of) a callgraph from one of my examples, you can see there are two examples in that digraph of methods defined on line -1:

![test_times_callgraph](https://cloud.githubusercontent.com/assets/97674/15568270/e781ff0a-2322-11e6-9eac-3a4d9ab98858.png)